### DISCUSSION TITLE
calling btagging module in analyses code in expert mode

### DISCUSSION THREAD

**Discussion by juhidutta** [MadAnalysis/madanalysis5#197] [Q&A]
Dear Experts,

Could you please kindly let me know how I may run the btagging on a jet from a lhco file in the analyses code for the expert mode  ?  

Best regards,
Juhi

**Comment by BFuks** [comment]
Hi Juhi,

This can be achieved through standard b-tagging methods: b-tagging information can be retrieved through the btag() method of the jet class (as described in the manual).

Cheers,

Benjamin

**Comment by juhidutta** [comment]
Dear Sir,

Thank you for your response.

Sorry for the confusion.

 I am referring to a case in which the lhco file only has jets without b
tag and I want to tag them separately in the code in expert mode. How could
one call the b tagging method in the code or does one have to check  for the
parent pdg code to be a B meson and if so,  tag the jet as a b jet?


With best regards,
Juhi

**Comment by BFuks** [comment]
Hi Juhi,

This information being not included in the LHCO file, this is impossible to do.

Regards,

Benjamin

**Comment by juhidutta** [comment]
Dear Sir,

Ok thank you for letting me know.

I have one last query pertaining to the latest madanalysis paper
https://inspirehep.net/literature/2639007.

On page 13, end of first paragraph, it mentions:

"Moreover, we define b jets as small jets with pT > 30 GeV and with a
ghost-associated b-hadron with pT > 5 GeV [57, 58]."

As far as I understood, in this case the hepmc file is used as an input and
fast jet is being used to cluster the fat jets and subsequently the small
jets.

 Could you please kindly elaborate on what the ghost associated b hadron
mean and  how I can access it from the hepmc file?

Thank you for your help,

With best regards,
Juhi

**Comment by BFuks** [comment]
Dear Juhi,

> As far as I understood, in this case the hepmc file is used as an input and
> fast jet is being used to cluster the fat jets and subsequently the small
> jets.
> 
Precisely. In this case, with the HepMC file we have all the hadron-level information so that we are able to implement any b-tagging method we wanted.

> Could you please kindly elaborate on what the ghost associated b hadron
> mean and how I can access it from the hepmc file?
We just implemented the algorithm as described in the references. As we have all the information, it is easy.

Cheers,

Benjamin

**Comment by juhidutta** [comment]
Dear Sir,

Thank you for your prompt response. I will look into the references.

With best regards,
Juhi
